packages/shared/src/components/modals/MarketingCtaModal.tsx

```diff
@@ -23,19 +23,23 @@ export const MarketingCtaModal = ({
 }: MarketingCtaModalProps): ReactElement => {
   const { clearMarketingCta } = useBoot();
   const { trackEvent } = useAnalyticsContext();
+  const { campaignId, flags } = marketingCta;
   const { tagColor, tagText, title, description, image, ctaUrl, ctaText } =
-    marketingCta.flags;
+    flags;
 
   const onModalClose = (
     param: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>,
     eventName: AnalyticsEvent,
   ) => {
-    trackEvent({
-      event_name: eventName,
-      target_type: TargetType.MarketingCtaPopover,
-      target_id: marketingCta.campaignId,
-    });
-    clearMarketingCta(marketingCta.campaignId);
+    if (campaignId) {
```

Comment:
since previously we were sending no matter `camaignId` can you confirm it is expected to only send in this case?

Reply:
Confirmed, we still need it here. Reverted this change.